renderdoc/driver/metal/metal_types.cpp

```diff
@@ -95,4 +97,114 @@ void DoSerialise(SerialiserType &ser, NS::String *&el)
   }
 }
 
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, MTL::TextureSwizzleChannels &el)
+{
+  SERIALISE_MEMBER(red);
+  SERIALISE_MEMBER(green);
+  SERIALISE_MEMBER(blue);
+  SERIALISE_MEMBER(alpha);
+}
+
+// MTLTextureDescriptor
+// {
+//   MTLTextureType textureType;
+//   MTLPixelFormat pixelFormat;
+//   NSUInteger width;
+//   NSUInteger height;
+//   NSUInteger depth;
+//   NSUInteger mipmapLevelCount;
+//   NSUInteger sampleCount;
+//   NSUInteger arrayLength;
+//   MTLResourceOptions resourceOptions;
+//   MTLCPUCacheMode cpuCacheMode
+//   MTLStorageMode storageMode
+//   MTLHazardTrackingMode hazardTrackingMode
+//   MTLTextureUsage usage
+//   BOOL allowGPUOptimizedContents
+//   MTLTextureSwizzleChannels swizzle
+// }
```

Comment:
Comments like this that you use for reference while writing the code should be removed.